test/HttpApiHandlerTests.cs

```diff
@@ -0,0 +1,123 @@
+            var totalTimeout = TimeSpan.FromSeconds(0);
+            var retryTimeout = TimeSpan.FromSeconds(100);
+            var ex = await Assert.ThrowsAsync<ArgumentException>(() => httpApiHandler.CreateCheckStatusResponse(new HttpRequestMessage() { RequestUri = new Uri(TestConstants.RequestUri) }, TestConstants.InstanceId, new OrchestrationClientAttribute() { TaskHub = TestConstants.TaskHub, ConnectionName = TestConstants.ConnectionName }, totalTimeout, retryTimeout));
+            Assert.Equal($"Total timeout {totalTimeout.TotalSeconds} should be bigger than retry timeout {retryTimeout.TotalSeconds}", ex.Message);
+        }
+
+        [Fact]
+        public async Task CreateCheckStatusResponse_Returns_Corrent_HTTP_202_Response()
+        {
+            var httpApiHandler = new HttpApiHandler(new DurableTaskExtension() { NotificationUrl = new Uri(TestConstants.NotificationUrl) }, null);
+            var httpResponseMessage = httpApiHandler.CreateCheckStatusResponse(new HttpRequestMessage() { RequestUri = new Uri(TestConstants.RequestUri) }, TestConstants.InstanceId, new OrchestrationClientAttribute() { TaskHub = TestConstants.TaskHub, ConnectionName = TestConstants.ConnectionName });
+            Assert.Equal(httpResponseMessage.StatusCode, HttpStatusCode.Accepted);
+            var content = await httpResponseMessage.Content.ReadAsStringAsync();
+            var status = JsonConvert.DeserializeObject<JObject>(content);
+            Assert.Equal(status["id"], TestConstants.InstanceId);
+            Assert.Equal(status["statusQueryGetUri"], "http://localhost:7071/admin/extensions/DurableTaskExtension/instances/7b59154ae666471993659902ed0ba742?taskHub=SampleHubVS&connection=Storage&code=mykey");
+            Assert.Equal(status["sendEventPostUri"], "http://localhost:7071/admin/extensions/DurableTaskExtension/instances/7b59154ae666471993659902ed0ba742/raiseEvent/{eventName}?taskHub=SampleHubVS&connection=Storage&code=mykey");
+            Assert.Equal(status["terminatePostUri"], "http://localhost:7071/admin/extensions/DurableTaskExtension/instances/7b59154ae666471993659902ed0ba742/terminate?reason={text}&taskHub=SampleHubVS&connection=Storage&code=mykey");
+        }
+
+        [Fact]
+
+        public async Task CreateCheckStatusResponse_Returns_HTTP_202_Response_After_Timeout()
+        {
+            var httpApiHandler = new HttpApiHandler(new DurableTaskExtensionMock() { NotificationUrl = new Uri(TestConstants.NotificationUrl) }, null);
+            var stopWatch = Stopwatch.StartNew();
+            var httpResponseMessage = await httpApiHandler.CreateCheckStatusResponse(new HttpRequestMessage() { RequestUri = new Uri(TestConstants.RequestUri) }, TestConstants.RandomInstanceId, new OrchestrationClientAttribute() { TaskHub = TestConstants.TaskHub, ConnectionName = TestConstants.ConnectionName }, TimeSpan.FromSeconds(100), TimeSpan.FromSeconds(10));
```

Comment:
Let's break this into multiple lines to avoid so much horizontal scrolling.

Reply:
Nice!